articles/node_lifecycle.md

```diff
@@ -27,224 +27,447 @@ Last Modified: {% if page.last_modified %}{{ page.last_modified }}{% else %}{{ p
+If a full cleanup is not possible, a user can elect to either return:
+- `FAILURE`: in which case the node will transition to `UncleanFinalized`
+- `ERROR`: in which case a runtime error will be thrown
 
-- If the `onShutdown` callback succeeds the node will transition to `Finalized`.
-- If the `onShutdown` callback raises or results in any other return code the node will transition to `ErrorProcessing`.
+*Outbound Transitions*
 
-### Transition State: ErrorProcessing
+- `SUCCESS` → `Unconfigured`
+- `FAILURE` → `UncleanFinalized`
+- `ERROR` → `throw runtime_error`
 
-This transition state is where any error can be cleaned up.
-It is possible to enter this state from any state where user code will be executed.
-If error handling is successfully completed the node can return to `Unconfigured`,
-If a full cleanup is not possible it must fail and the node will transition to `Finalized` in preparation for destruction.
 
-Transitions to `ErrorProcessing` may be caused by error return codes in callbacks as well as methods within a callback or an uncaught exception.
+#### **Transition State: `ShuttingDown`**
+This state is expected to do any cleanup necessary before destruction.
+It may be entered from any Primary State except terminal states (i.e., `Finalized` and `UncleanFinalized`).
 
-#### Valid transitions out of ErrorProcessing
+*Outbound Transitions*
+- `SUCCESS` → `Finalized`
+- `FAILURE` → `prior_state`: {`Unconfigured`, `Active`, `Inactive`}
+- `ERROR` → `ErrorProcessing`
 
-- If the `onError` callback succeeds the node will transition to `Unconfigured`.
-  It is expected that the `onError` will clean up all state from any previous state.
-  As such if entered from `Active` it must provide the cleanup of both `onDeactivate` and `onCleanup` to return success.
+#### **Operation `create()`**
+This operation will instantiate the node, but will not run any code beyond the constructor.
 
-- If the `onShutdown` callback raises or results in any other result code the node will transition to `Finalized`.
+#### **Operation `destroy()`**
+This is a stateless operation to be taken from the `Finalized` state.
+This operation will simply cause the deallocation of the node.
+This operation should always succeed if taken from the `Finalized` state.
 
-### Destroy Transition
+#### **Operation `raise_error()`**
+This operation transitions the node from the primary states `{Unconfigured, Inactive, Active}` to `ErrorProcessing`.
+This is used to attempt a recovery to `Unconfigured` from a Primary State in the case of an internal error.
+Thus, this operation is only available internally to the node.
 
-This transition will simply cause the deallocation of the node.
-In an object oriented environment it may just involve invoking the destructor.
-Otherwise it will invoke a standard deallocation method.
-This transition should always succeed.
 
-### Create Transition
+### Return Values
```

Comment:
I get that for transition states, but is that true for primary states as well? So no exception catching to cause us to go into error processing if an uncaught exception is thrown in a primary state? I think that's untrue given how the primary states work, so I think this needs clarification that we're only referring to transition states.